src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisExerciseChatSessionService.java

```diff
@@ -170,10 +171,26 @@ public void handleStatusUpdate(ExerciseChatJob job, PyrisChatStatusUpdateDTO sta
             var message = new IrisMessage();
             message.addContent(new IrisTextMessageContent(statusUpdate.result()));
             var savedMessage = irisMessageService.saveMessage(message, session, IrisMessageSender.LLM);
-            irisChatWebsocketService.sendMessage(savedMessage, statusUpdate.stages());
+            irisChatWebsocketService.sendMessage(savedMessage, statusUpdate.stages(), statusUpdate.suggestions());
         }
         else {
             irisChatWebsocketService.sendStatusUpdate(session, statusUpdate.stages());
         }
+
+        if (statusUpdate.suggestions() != null) {
+            updateLatestSuggestions(session, statusUpdate.suggestions());
+        }
+    }
+
+    /**
+     * Updates the latest suggestions of the session.
+     *
+     * @param session           The session to update
+     * @param latestSuggestions The latest suggestions to set
+     */
+    private void updateLatestSuggestions(IrisExerciseChatSession session, List<String> latestSuggestions) {
```

Comment:
duplicated method.

Reply:
Refactored and moved the method to an abstract class